# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: architecture/powerpc64/package/.../0180-lv1call-add-undocumented-spe-hvcalls.patch
# Copyright (C) 2019 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- a/arch/powerpc/include/asm/lv1call.h
+++ b/arch/powerpc/include/asm/lv1call.h
@@ -41,6 +41,7 @@
 #define LV1_5_IN_0_OUT_ARG_DECL LV1_5_IN_ARG_DECL
 #define LV1_6_IN_0_OUT_ARG_DECL LV1_6_IN_ARG_DECL
 #define LV1_7_IN_0_OUT_ARG_DECL LV1_7_IN_ARG_DECL
+#define LV1_8_IN_0_OUT_ARG_DECL LV1_8_IN_ARG_DECL
 
 #define LV1_0_IN_1_OUT_ARG_DECL                    LV1_1_OUT_ARG_DECL
 #define LV1_1_IN_1_OUT_ARG_DECL LV1_1_IN_ARG_DECL, LV1_1_OUT_ARG_DECL
@@ -131,6 +132,7 @@
 #define LV1_5_IN_0_OUT_ARGS LV1_5_IN_ARGS
 #define LV1_6_IN_0_OUT_ARGS LV1_6_IN_ARGS
 #define LV1_7_IN_0_OUT_ARGS LV1_7_IN_ARGS
+#define LV1_8_IN_0_OUT_ARGS LV1_8_IN_ARGS
 
 #define LV1_0_IN_1_OUT_ARGS                LV1_1_OUT_ARGS
 #define LV1_1_IN_1_OUT_ARGS LV1_1_IN_ARGS, LV1_1_OUT_ARGS
@@ -241,6 +243,7 @@ LV1_CALL(shutdown_logical_partition,                    1, 0,  44 )
 LV1_CALL(destruct_logical_spe,                          1, 0,  54 )
 LV1_CALL(construct_logical_spe,                         7, 6,  57 )
 LV1_CALL(set_spe_interrupt_mask,                        3, 0,  61 )
+LV1_CALL(undocumented_function_62,                      5, 0,  62 )
 LV1_CALL(set_spe_transition_notifier,                   3, 0,  64 )
 LV1_CALL(disable_logical_spe,                           2, 0,  65 )
 LV1_CALL(clear_spe_interrupt_status,                    4, 0,  66 )
@@ -257,6 +260,7 @@ LV1_CALL(write_repository_node,                         7, 0,  92 )
 LV1_CALL(delete_repository_node,                        5, 0,  93 )
 LV1_CALL(read_htab_entries,                             2, 5,  95 )
 LV1_CALL(set_dabr,                                      2, 0,  96 )
+LV1_CALL(undocumented_function_99,                      2, 0,  99 )
 LV1_CALL(get_total_execution_time,                      2, 1, 103 )
 LV1_CALL(undocumented_function_114,                     3, 1, 114 )
 LV1_CALL(undocumented_function_115,                     1, 0, 115 )
@@ -267,12 +271,15 @@ LV1_CALL(destruct_io_irq_outlet,                        1, 0, 121 )
 LV1_CALL(map_htab,                                      1, 1, 122 )
 LV1_CALL(unmap_htab,                                    1, 0, 123 )
 LV1_CALL(get_version_info,                              0, 2, 127 )
+LV1_CALL(undocumented_function_138,                     2, 0, 138 )
 LV1_CALL(insert_htab_entry,                             6, 3, 158 )
 LV1_CALL(read_virtual_uart,                             3, 1, 162 )
 LV1_CALL(write_virtual_uart,                            3, 1, 163 )
 LV1_CALL(set_virtual_uart_param,                        3, 0, 164 )
 LV1_CALL(get_virtual_uart_param,                        2, 1, 165 )
 LV1_CALL(configure_virtual_uart_irq,                    1, 1, 166 )
+LV1_CALL(undocumented_function_167,                     2, 1, 167 )
+LV1_CALL(undocumented_function_168,                     3, 0, 168 )
 LV1_CALL(open_device,                                   3, 0, 170 )
 LV1_CALL(close_device,                                  2, 0, 171 )
 LV1_CALL(map_device_mmio_region,                        5, 1, 172 )
@@ -293,8 +300,11 @@ LV1_CALL(net_control,                                   6, 2, 194 )
 LV1_CALL(connect_interrupt_event_receive_port,          4, 0, 197 )
 LV1_CALL(disconnect_interrupt_event_receive_port,       4, 0, 198 )
 LV1_CALL(get_spe_all_interrupt_statuses,                1, 1, 199 )
+LV1_CALL(undocumented_function_200,                     1, 0, 200 )
+LV1_CALL(undocumented_function_201,                     1, 0, 201 )
 LV1_CALL(deconfigure_virtual_uart_irq,                  0, 0, 202 )
 LV1_CALL(enable_logical_spe,                            2, 0, 207 )
+LV1_CALL(undocumented_function_209,                     8, 0, 209 )
 LV1_CALL(gpu_open,                                      1, 0, 210 )
 LV1_CALL(gpu_close,                                     0, 0, 211 )
 LV1_CALL(gpu_device_map,                                1, 2, 212 )
--- a/arch/powerpc/platforms/ps3/hvcall.S
+++ b/arch/powerpc/platforms/ps3/hvcall.S
@@ -33,6 +33,7 @@ _GLOBAL(_##API_NAME)				\
 #define LV1_5_IN_0_OUT LV1_N_IN_0_OUT
 #define LV1_6_IN_0_OUT LV1_N_IN_0_OUT
 #define LV1_7_IN_0_OUT LV1_N_IN_0_OUT
+#define LV1_8_IN_0_OUT LV1_N_IN_0_OUT
 
 #define LV1_0_IN_1_OUT(API_NAME, API_NUMBER)	\
 _GLOBAL(_##API_NAME)				\
